airflow/jobs.py

```diff
@@ -212,27 +212,48 @@ def _execute(self):
         raise NotImplementedError("This method needs to be overridden")
 
     @provide_session
-    def reset_state_for_orphaned_tasks(self, dag_run, session=None):
+    def reset_state_for_orphaned_tasks(self, dag_run=None, session=None):
```

Comment:
Not the biggest fan of changing the function behavior based on the parameter instead of just using multiple functions as you had before, e.g. separation of concerns, and the current implementation is not very intuitive since the parameter is called "dag_run" which implies it only affects a single dag run. Not a blocker though.